models/note.rb

```diff
@@ -0,0 +1,25 @@
+class Note < ActiveRecord::Base
+  has_many :taggings
+  has_many :tags, through: :taggings
+  validates :body, presence: true
+  validates :title, presence: true
+
+
+
+  def example_note
+    {
+      "title"       => title,
+      "body"        => body,
+      "tags"        => tags.map { |t| {"name" => t.name} }
+    }
+  end
+
+  def as_json(arg = nil)
```

Comment:
Since you defined this, your controller action might have needed less code. `@notes.to_json` might have worked just fine.